projects/website/src/doxygen/XSDToDoxygen.py

```diff
@@ -0,0 +1,284 @@
+    """
+    Process_sc
+    _____________
+
+    This function stores all of the files within the xs:schema parent node 
+    and then parses through all files and then append the file names into files_processed list
+
+    """
+
+    def process_sc(self,fout):
+        for input_file in self.files_processed: 
+            sc=minidom.parse(input_file)
+            for child in sc.childNodes:
+                if child.nodeName == "xs:schema":
+                    self.parse(child,fout)
+
+        """
+    Parser
+    ___________
+
+    This is the main function which is called again and again till childnodes of the heirarchy is found
+    It tries to align those heirarchial data structure according to comments and other nodes which play 
+    certain crucial role in parsing
+    Nodes are:-
+    xs:simpletype
+    xs:complextype
+    xs:extension
+    xs:restriction
+    xs:enumeration
+    xs:element
+    xs:sequence
+    xs:attribute
+    xs:include
+    and 
+    node Type namely Comment_node
+    """
+
+    def parse(self,node,fout):
+        """
+        For parsing through all respective child Nodes in a XSD file
+        """
+        for j,i in enumerate(node.childNodes):
+            
+            """
+            Case 1: Node Name = Simple Type or Complex Type
+            Extract attribute name from the nodes
+            Then append that into list of name
+            And also write the brief related comments from the file
+            """
+
+            if i.nodeName == "xs:simpleType" or i.nodeName == "xs:complexType":
+                if i.getAttribute("name"):
+                    self.name.append(i.getAttribute("name"))
+                    fout.write(" * @defgroup CDM_"+i.getAttribute("name")+" "+i.getAttribute("name"))
+                    fout.write("\n")
+                    if len(self.writer)>0:
+                        fout.write(self.writer)
+                        self.writer=""
+                else:
+                    self.name.append("null")
+                    fout.write(" * @defgroup CDM_"+self.name[-3]+"_"+self.name[-2]+"_"+self.name[-1]+" "+self.name[-1])
+                    fout.write("\n")
+                    if len(self.writer)>0:
+                        fout.write(self.writer)
+                        self.writer=""
+               
+                """
+                Reitrating through the process for all of the subsequent child nodes.
+                For names if name is attribute loop will only function once 
+                And only once the parse function will run
+                """
+                
+                names_1=[s.nodeName for s in i.childNodes if s.nodeName in self.tags or s.nodeType==s.COMMENT_NODE]
+                if self.tags[7] in list(set(names_1)):
+                    for l in i.childNodes:
+                        if l.nodeName in self.tags or l.nodeType==l.COMMENT_NODE:
+                            fout.write(" * @{")
+                            fout.write("\n")
```

Comment:
Move the return line to the first write command no need to call write twice. Do this through out the file

Reply:
You meant I need to create a function for the writing part?